astrowidgets/core.py

```diff
@@ -443,16 +455,28 @@ def get_markers(self, x_colname='x', y_colname='y',
             Table of markers, if any, or ``None``.
 
         """
+        if marker_name is None:
+            marker_name = self._default_mark_tag_name
+
+        if marker_name == 'all':
+            tables = [self.get_markers(x_colname=x_colname,
+                                       y_colname=y_colname,
+                                       skycoord_colname=skycoord_colname,
+                                       marker_name=name)
+                      for name in self._marktags]
+            return vstack(tables, join_type='exact')
+
         try:
-            c_mark = self._viewer.canvas.get_object_by_tag(self._marktag)
+            c_mark = self._viewer.canvas.get_object_by_tag(marker_name)
         except Exception as e:  # No markers
             self.logger.warning(str(e))
             return
 
         image = self._viewer.get_image()
         xy_col = []
 
-        if image.wcs.wcs is None:  # Do not include SkyCoord column
+        if (image is None) or (image.wcs.wcs is None):
```

Comment:
Why is `image is None` comparison needed here? One cannot mark on an empty image widget, so at this point of logic, the image has to exist, no?

Reply:
In practice one *should* always have an image, but it is possible with the current API to add markers in the absence of an image. They don't display, since display only happens when there is an image. I think this came up when I was writing tests...